src/HttpClient/Plugin/GzipEncoderPlugin.php

```diff
@@ -39,8 +48,14 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
             $requestBody->rewind();
         }
 
+        $encodedBody = gzcompress($requestBody->getContents(), -1, ZLIB_ENCODING_GZIP);
+
+        if (false === $encodedBody) {
+            throw new \RuntimeException('Failed to GZIP-encode the request body.');
```

Comment:
Wouldn't this make the app blow up?

Reply:
No because we catch any and all errors occurring in the transport and discard them to prevent the blowing up part.

Reply:
Wouldn't this make the app blow up?

It would, and since it's an error happening while sending the event I don't have a good reason we should not thrown an exception. In case users are using our `HttpTransport` the exception will be caught and silently discarded though...